modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -1558,16 +1559,53 @@ private void _validateListTypeEntriesValues(
 		}
 	}
 
+	private void _validateObjectFieldIntegerTypeSize(
+			Map.Entry<String, Serializable> entry)
+		throws ObjectEntryValuesException {
+
+		Serializable entryValue = entry.getValue();
+
+		if (!StringUtil.equals(entryValue.toString(), "")) {
+			int value = GetterUtil.getInteger(entryValue);
+
+			if (!StringUtil.equals(
+					String.valueOf(value), entryValue.toString())) {
+
+				throw new ObjectEntryValuesException.IntegerMaxSizeExceeded();
+			}
+		}
+	}
+
+	private void _validateObjectFieldLongType(
+			Map.Entry<String, Serializable> entry)
+		throws ObjectEntryValuesException {
+
+		Serializable entryValue = entry.getValue();
+
+		if (!StringUtil.equals(entryValue.toString(), "")) {
+			long value = GetterUtil.getLong(entryValue);
+
+			if (!StringUtil.equals(
+					String.valueOf(value), entryValue.toString())) {
+
+				throw new ObjectEntryValuesException.LongMaxSizeExceeded();
+			}
+			else if ((value > ObjectEntryConstants.MAX_SAFE_LONG) ||
```

Comment:
Please, create separate conditions for these cases with the specific messages (max and min).